src/integrationTest/java/uk/gov/hmcts/darts/usermanagement/controller/SecurityRoleControllerIntTest.java

```diff
@@ -0,0 +1,107 @@
+package uk.gov.hmcts.darts.usermanagement.controller;
+
+import org.junit.jupiter.api.Test;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import uk.gov.hmcts.darts.authorisation.component.UserIdentity;
+import uk.gov.hmcts.darts.testutils.IntegrationBase;
+import uk.gov.hmcts.darts.testutils.stubs.AdminUserStub;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@AutoConfigureMockMvc
+class SecurityRoleControllerIntTest extends IntegrationBase {
```

Comment:
Could you add an int test to cover AC2 "I have obtained an access token as a non-ADMIN user, and the security_roles table is populated." please?

Reply:
ok, although that would be testing more the Authorisation code and not my endpoint per-se. I.e. if authorisation throws a different HTTP response code, then its not really in the remit of this ticket to change that

Reply:
Purpose is to verify that the correct access rights are applied to the controller method via the `@Authorisation` annotation. I don't think we have any other mechanism to verify that only admins can access admin endpoints, on an endpoint-by-endpoint basis. If the `@Authorisation` annotation is ever accidently removed, then the test should catch that.